org.mixedrealitytoolkit.input/Tests/Runtime/BasicInputTests.cs

```diff
@@ -592,6 +693,46 @@ public IEnumerator UntrackedControllerNearInteractions()
             yield return null;
         }
 
+        /// <summary>
+        /// Test the HandModel script has the required fields.
+        /// </summary>
+        [UnityTest]
+        public IEnumerator HandModelHasRequiredFieldsAndAccessors()
+        {
+            FieldInfo[] fieldInfos;
+            PropertyInfo[] accessorsInfos;
+            Type HandModel = typeof(HandModel);
+
+            fieldInfos = HandModel.GetFields(BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Public);
+            accessorsInfos = HandModel.GetProperties(BindingFlags.Instance | BindingFlags.Public);
+
+            var modelPrefabFieldInfo = fieldInfos.Where(fieldInfo => fieldInfo.Name.Equals("modelPrefab")).ToArray();
+            Assert.AreEqual(1, modelPrefabFieldInfo.Length, "HandModel is missing the 'modelPrefab' field");
+
+            var modelParentFieldInfo = fieldInfos.Where(fieldInfo => fieldInfo.Name.Equals("modelParent")).ToArray();
+            Assert.AreEqual(1, modelParentFieldInfo.Length, "HandModel is missing the 'modelParent' field");
+
+            var modelFieldInfo = fieldInfos.Where(fieldInfo => fieldInfo.Name.Equals("model")).ToArray();
+            Assert.AreEqual(1, modelFieldInfo.Length, "HandModel is missing the 'model' field");
+
+            var handNodeFieldInfo = fieldInfos.Where(fieldInfo => fieldInfo.Name.Equals("handNode")).ToArray();
+            Assert.AreEqual(1, modelFieldInfo.Length, "HandModel is missing the 'handNode' field");
+
+            var modelPrefabAccessorInfo = accessorsInfos.Where(accessorInfo => accessorInfo.Name.Equals("ModelPrefab")).ToArray();
+            Assert.AreEqual(1, modelPrefabAccessorInfo.Length, "HandModel is missing the 'ModelPrefab' accessor");
+
+            var modelParentAccessorInfo = accessorsInfos.Where(accessorInfo => accessorInfo.Name.Equals("ModelParent")).ToArray();
+            Assert.AreEqual(1, modelParentAccessorInfo.Length, "HandModel is missing the 'ModelParent' accessor");
+
+            var modelAccessorInfo = accessorsInfos.Where(accessorInfo => accessorInfo.Name.Equals("Model")).ToArray();
+            Assert.AreEqual(1, modelAccessorInfo.Length, "HandModel is missing the 'Model' accessor");
+
+            var handNodeAccessorInfo = accessorsInfos.Where(accessorInfo => accessorInfo.Name.Equals("HandNode")).ToArray();
+            Assert.AreEqual(1, handNodeAccessorInfo.Length, "HandModel is missing the 'HandNode' accessor");
+
```

Comment:
What purpose does this test serve?  It is only testing the field and property layout of this component, not functionality.  This doesn't appear to be a standard type of test we typically write.

Reply:
I see no harm in this.   I guess it could catch mistakenly deleted components.

Reply:
I thought on this test aiming to ensure the deprecated XRController Model functionality is available in the XRI3 migrated code.  Also, I recall one bug in which there was a regression because of class having a field removed so this could help prevent that.  It is a "HandModel does have what it should" kind of test.  Resolving.